.github/workflows/cd.yml

```diff
@@ -26,14 +29,16 @@ jobs:
           GH_REPO: ${{ github.repository }}
         # We use the `gh` utility as it offers much easier traversal of GitHub-
         # related things (prs, issues, etc...).
+        # In this merged_branch variable, we get the list of PRs that have been merged,
+        # and sort them by when they were merged, taking the most recent (last) and
+        # returning the branch name of that PR.
         # The `cut` command splits the `pre-<version>` branch into just `<version>`,
         # which will be the new tag.
         run: |
           merged_branch=$(gh pr list \
             --state merged \
-            --limit 1 \
-            --json 'headRefName' \
-            --jq '.[].headRefName')
+            --json 'headRefName,mergedAt' \
+            --jq 'sort_by(.mergedAt) | last | .headRefName')
```

Comment:
I ran this on this repo. 

```
$ gh pr list --state merged

Showing 11 of 11 pull requests in ACCESS-NRI/ACCESS-OM2 that match your search

#36  PR now notifies you of `spack-{packages,config}` version about to be deployed  32-ci-versions-comment       about 1 day ago
#35  `Prerelease` Functionaility                                                    34-prerelease                about 5 days ago
#31  Require parallelio 2.5.2                                                       30-require-parallelio-2.5.2  about 13 days ago
#29  Modulefile name declarations                                                   26-modulefile-additions      about 1 month ago
#28  GitHub Release Provenance                                                      24-gh-release-provenance     about 2 months ago
#27  Update README.md                                                               aidanheerdegen/fix-README    about 2 months ago
#23  YAML Reorganisation                                                            22-yaml-reorganisation       about 2 months ago
#21  Updated README.md                                                              update-readme                about 2 months ago
#18  Added explicit compilers to spack spec                                         explicit-compilers           about 2 months ago
#17  More Explicit Versions and MOM5 update                                         explicit-versioning          about 2 months ago
#16  spack.yaml: Added oasis3-mct as a versioned dependency                         access-om2-update            about 2 months ago
```

But the command used in the CI returns the PR that was merged 5 days ago, not the most recent merge.
```
$ gh pr list --state merged --json 'headRefName,mergedAt' --jq 'sort_by(.mergedAt) | last | .headRefName'
34-prerelease
```

Reply:
I don't think it was merged 5 days ago, it was created 5 days ago. We just merged that one today. If you run:
`gh pr list --state merged --json 'headRefName,mergedAt' --jq 'sort_by(.mergedAt) | last'` it should note that it was merged today.